File: Easy/2942. Find Words Containing Character/Go/main.go
func findWordsContaining(words []string, x byte) []int {
    var result []int

    for i, word := range words {
      if containsChar(word, x) {
        result = append(result, i)
      }
    }

    return result
}

func containsChar(word string, x byte) bool {
  for i := 0; i < len(word); i++ {
    if word[i] == x {
      return true
    } 
  }
  return false
}

File: Easy/2942. Find Words Containing Character/Rust/main.rs
impl Solution {
    pub fn find_words_containing(words: Vec<String>, x: char) -> Vec<i32> {
        let mut result = Vec::new();

        for (i, word) in words.iter().enumerate() {
            println!("Checking word at index {}: \"{}\"", i, word);

            if word.contains(x) {
                println!("  → Contains '{}', adding index {}", x, i);
                result.push(i as i32);
            } else {
                println!("  → Does not contain '{}'", x);
            }
        }

        println!("Final result: {:?}", result);
        result
    }
}

